src/gwt/src/org/rstudio/core/client/widget/TextBoxWithButton.java

```diff
@@ -95,6 +96,10 @@ public HandlerRegistration addValueChangeHandler(
       return addHandler(handler, ValueChangeEvent.getType());
    }
   
+   public HandlerRegistration addTextBoxValueChangeHandler(ChangeHandler handler)
```

Comment:
Makes sense, looked a bit into this, the current behavior is to trigger a value change ONLY when setText is called:

```
public void setText(String text)
   {
      text_ = text;
      
      if (text_.equals(useDefaultValue_))
         textBox_.setText("[Use Default] " + text);
      else if (text.length() > 0)
         textBox_.setText(text);
      else
         textBox_.setText(emptyLabel_);
      
      ValueChangeEvent.fire(this, getText());
   }
```

Currently, setText() is only triggered through the browse button not with changes performed directly in the textbox. Instead of changing the base class, I'm subclassing this with a new FileOrUrlChooserTextBox widget.